pkg/operator/certrotation/signer.go

```diff
@@ -87,7 +87,9 @@ func (c RotatedSigningCASecret) EnsureSigningCertKeyPair(ctx context.Context) (*
 
 	// apply necessary metadata (possibly via delete+recreate) if secret exists
 	// this is done before content update to prevent unexpected rollouts
-	if ensureMetadataUpdate(signingCertKeyPairSecret, c.Owner, c.AdditionalAnnotations) && ensureSecretTLSTypeSet(signingCertKeyPairSecret) {
+	needsMetadataUpdate := ensureMetadataUpdate(signingCertKeyPairSecret, c.Owner, c.AdditionalAnnotations)
+	needsTypeUpdate := ensureSecretTLSTypeSet(signingCertKeyPairSecret)
```

Comment:
Why does ensureSecretTLSTypeSet have to mess with the content of the secret?

xref: https://github.com/openshift/library-go/blob/0b298992f8cc44a4dc612800bd6766acdba97008/pkg/operator/certrotation/metadata.go#L18

Reply:
`ensureSecretTLSTypeSet` may change secret type if its not `kubernetes.io/tls`